pkg/tsdb/prometheus/querydata/response.go

```diff
@@ -29,6 +29,11 @@ func (s *QueryData) parseResponse(ctx context.Context, q *models.Query, res *htt
 		MatrixWideSeries: s.enableWideSeries,
 		VectorWideSeries: s.enableWideSeries,
 	})
+
+	if r.Error != nil {
+		return r, r.Error
+	}
+
```

Comment:
I believe we should add a frame with no fields (if there isn't one already), and return after adding the metadata to the frame (So interpolated queries are still returned). See para 2 of https://github.com/grafana/grafana-plugin-sdk-go/tree/main/data/contract_docs#error-responses : 

> When an error is returned with the DataResponse, a single frame with no fields may be included as well. If the error is present, this will not be considered "No Data". This frame is included so that metadata, in particular a Frame's ExecutedQueryString, can be returned to the caller with an error.

Reply:
I think we always have a frame in it. See https://github.com/grafana/grafana/pull/57209/files#r999603684